### PR TITLE
DC-3075: Get CDS conversations (list) API on secure-message

### PR DESCRIPTION
**Pull request by matthewgroom**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -80,7 +80,6 @@ lazy val microservice = Project(appName, file("."))
       "-Xlint:unsound-match", // Pattern match may not be typesafe.
       "-Yno-adapted-args", // Do not adapt an argument list (either by inserting () or creating a tuple) to match the receiver.
       "-Ypartial-unification", // Enable partial unification in type constructor inference
-      "-Ywarn-dead-code", // Warn when dead code is identified.
```
Comment (lxol) [inline build.sbt:83]: wasn't it possible to delete dead code instead?

Comment (matthewgroom) [reply build.sbt:83]: I get this error in my tests when mocking the authorise call => [error] /home/temp/Applications/hmrc-development-environment/hmrc/secure-message/test/uk/gov/hmrc/securemessage/controllers/SecureMessageControllerSpec.scala:133:43: dead code following this construct
[error]       when(mockAuthConnector.authorise(any(), any[Retrieval[Enrolments]])(any(), any()))

File: app/uk/gov/hmrc/securemessage/controllers/models/generic/ConversationDetails.scala [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      .find(_.identifier === identifier)
+      .fold(false)(participant => findUnreadMessagesByParticipant(participant.id, coreConversation))
+  }
+
+  private def findUnreadMessagesByParticipant(participantId: Int, coreConversation: Conversation): Boolean = {
+    val messages = coreConversation.messages
+    val readers = messages.flatMap(ms => ms.readBy)
+    val readerIds = readers.map(reader => reader.id)
+    val matchingIds = readerIds.count(int => int === participantId)
+    matchingIds =!= messages.size
```
Comment (elabeca) [inline app/uk/gov/hmrc/securemessage/controllers/models/generic/ConversationDetails.scala:79]: left a private message about that one

File: app/uk/gov/hmrc/securemessage/controllers/models/generic/ConversationDetails.scala [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    coreConversation.messages.sortWith(_.created.getMillis > _.created.getMillis).headOption match {
+      case Some(message) => Some(message)
+      case _             => None
+    }
+
+  private def findLatestMessageDate(coreConversation: Conversation): Option[DateTime] =
+    findLatestMessage(coreConversation) match {
+      case Some(ms) => Some(ms.created)
+      case _        => None
+    }
```
Comment (rwalpole) [inline app/uk/gov/hmrc/securemessage/controllers/models/generic/ConversationDetails.scala:59]: You can `flatMap` this as follows:
` findLatestMessage(coreConversation).flatMap(ms => Some(ms.created))`

**Review by elabeca — APPROVED** [review]

File: app/uk/gov/hmrc/securemessage/controllers/models/generic/ConversationDetails.scala (hunk 3 above)
Comment (matthewgroom) [reply app/uk/gov/hmrc/securemessage/controllers/models/generic/ConversationDetails.scala:58]: Thanks, Robertos!

**Review by elabeca — APPROVED** [review]

**Pull request merged**